### ISSUE TITLE
Original plugin gets deleted, not just the copy for the wp-now instance

### ISSUE THREAD

**Issue by GaryJones** [WordPress/playground-tools#29]
## Summary

Starting `wp-now` in a plugin directory, and then deleting that plugin via the UI, deletes the code in the original directory (i.e. local git repo) as well as the copy of the code in the `~/.wp-now/wp-content/...` directory.

## Steps to Reproduce

1. Go to a local plugin directory (e.g. `~/code/my-plugin`) and run `wp-now start`. wp-now starts up in _plugin_ mode, and opens a WP instance front page in a browser.
2. Navigate to the Plugins page and delete your local deactivate plugin (e.g. My Plugin) using the red “Delete” link.
3. It will claim to not fully delete (see screenshot below).
a. Note that `~/.wp-now/wp-content/plugins/my-plugin-{hash}/plugins/my-plugin/` still exists, but is empty. ✅ 
b. Note that the local plugin directory (`~/code/my-plugin`) still exists, but is also empty. 😱 

<img width="789" alt="Screenshot 2023-05-16 at 17 47 28" src="https://github.com/WordPress/wordpress-playground/assets/88371/ef134adf-c632-40dd-9d16-a850a44fac3e">

## Expected Behaviour
My expected behaviour would be that the copy of my-plugin in `~/.wp-now` would be deleted (per the active WP Now instance), but my original code (elsewhere in my system) would be left alone – especially if I was just wanting to test an `uninstall.php`, for instance.

----

## Aside

Trying to start wp-now in a plugin directory that is (now!) empty has some non-friendly error handling (it tried starting in index mode):

<img width="1598" alt="Screenshot 2023-05-16 at 17 49 41" src="https://github.com/WordPress/wordpress-playground/assets/88371/eedb47b3-9331-4281-9ffa-e925e4710309">

**Comment by danielbachhuber** [comment]
Thanks for the report, @GaryJones !

> My expected behaviour would be that the copy of my-plugin in `~/.wp-now` would be deleted (per the active WP Now instance), but my original code (elsewhere in my system) would be left alone – especially if I was just wanting to test an `uninstall.php`, for instance.

This is a really interesting assumption. `wp-now` doesn't copy the current working directory to `~/.wp-now`, but I can see how you might think about that.

We'll have to consider what might be the most appropriate UX for this scenario. I think it's preferable to continue to mount the current working directory into the virtual filesystem so any file changes are applied to the site. We could potentially add some custom code to prevent the plugin from being deleted in 'plugin' mode.

> especially if I was just wanting to test an `uninstall.php`, for instance.

This is useful context, thank you.

> Trying to start wp-now in a plugin directory that is (now!) empty has some non-friendly error handling (it tried starting in index mode):

I agree. This will be fixed with https://github.com/WordPress/wordpress-playground/pull/331

**Comment by GaryJones** [comment]
> We could potentially add some custom code to prevent the plugin from being deleted in 'plugin' mode.

That would work - hide the delete link for the plugin/theme that was mounted. At the very least, include an AYS with a warning that this will delete the original source of the (mounted) code.

**Comment by squelchdesign** [comment]
I agree with @GaryJones: I also wouldn't expect wp-now to delete my repo when I uninstall the plugin inside of WordPress. But I also would want to be able to make live changes to my source and have those changes reflected immediately without having to restart wp-now. Personally I wouldn't want wp-now to _ever_ change anything in my plugin repo, but I would want it to see changes that **I** make to the repo.

I noticed in the docs the suggestion of having WP_DEBUG_LOG point to the plugin/theme directory:

> If you prefer to set a custom path for the debug log file, you can set WP_DEBUG_LOG to be a directory. Remember that the php-wasm server runs udner a VFS (virtual file system) where the default documentRoot is always /var/www/html.
> 
> For example, if you run wp-now start --blueprint=path/to/blueprint-example.json from a theme named atlas you could use this directory: /var/www/html/wp-content/themes/atlas/example.log and you will find the example.log file in your project directory.

I can certainly see how that would be useful, but I'm of the opinion that my code is for me and only me to modify — WordPress / wp-now / playground shouldn't be able to modify it unless I have explicitly stated that's okay to do. I'd personally love to see the plugin/theme treated as read-only by wp-now, with perhaps a command-line switch or config option that provides permission to modify specific paths for logging etc (or, for those who like to live dangerously, they could specify that the entire plugin/theme is fair game).
